### ISSUE TITLE
Handle serialization of cyclic object

### ISSUE THREAD

**Issue by dnepro** [DataDog/browser-sdk#807]
Hey there!

I work with a very large Angular Application which creates a an Error like:
```
11:02:55.366 ERROR Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: 'collectionSize: 0'. Current value: 'collectionSize: 4'.
    Angular (7)
    View_FlightTimeListViewComponent_0 FlightTimeListViewComponent.html:21
    Angular (66)
    RxJS (6)
    Angular (9)
automaticErrorCollection.js:49
```
followed by a second error like:
```
11:02:55.377 ERROR CONTEXT 
{…}
​
elDef: Object { nodeIndex: 29, bindingIndex: 3, outputIndex: 0, … }
​
elView: Object { def: {…}, parent: {…}, state: 1036, … }
​
nodeDef: Object { nodeIndex: 30, bindingIndex: 3, outputIndex: 1, … }
​
nodeIndex: 30
​
view: Object { def: {…}, parent: {…}, state: 1036, … }
​
<prototype>: Object { elOrCompView: Getter, injector: Getter, component: Getter, … }
automaticErrorCollection.js:49
```

the first error shows up correctly in datadog, but the second results in:
`console error: ERROR CONTEXT <error: unable to serialize object>`

When i tried to serialize the error with "JSON.stringify", i receive this: 
```
11:21:16.362 Uncaught TypeError: cyclic object value
    <anonymous> debugger eval code:1
debugger eval code:1:6
```
It would be great if something like from this [MDN Article on TypeError: cyclic object value](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Errors/Cyclic_object_value) could be added directly to the browser logs or could be passed to datadog

Best Regards,
Roman

**Comment by dnepro** [comment]
I just found the [createCircularReferenceChecker](https://github.com/DataDog/browser-sdk/blob/master/packages/core/src/tools/context.ts#L53) but it seems it is not used or has some other issues

**Comment by bcaudan** [comment]
Hello @dnepro,

Indeed that is our current behavior when trying to serialize an object with a cycle: cf [related test](https://github.com/DataDog/browser-sdk/blob/master/packages/core/src/tools/utils.spec.ts#L262).
Removing the cyclic property from the serialization could be a nice improvement.
We'd be happy to review a PR for it if you want to give it a try.

**Comment by yannickadam** [comment]
Hi!
Serialization now removes cyclic references, available from SDK 4.23.3.
If you experience any issue, feel free to re-open this issue.
